packages/engine/lib/orchestrator/src/experiment.rs

```diff
@@ -95,6 +95,25 @@ pub struct ExperimentConfig {
     clap(global = true, short = 'w', long, default_value_t = num_cpus::get(), validator = at_least_one, env = "HASH_WORKERS")
     )]
     pub num_workers: usize,
+
+    #[cfg_attr(feature = "clap", clap(global = true, long, default_value_t = 0))]
+    /// Initial size of the V8 heap.
+    ///
+    /// Setting this value allows to avoid garbage collection while the heap is small enough.
+    pub v8_initial_heap_constraint: usize,
+
+    #[cfg_attr(
+        feature = "clap",
+        clap(global = true, long, default_value_t = 4_000_000_000)
+    )]
+    /// Max size of the V8 heap.
+    ///
+    /// V8 will run a series of garbage collection when the heap size gets close to this limit.
```

Comment:
This sounds as V8 will not try to run the GC unless this limit is reached.

Reply:
What do you propose? This is close to rusty_v8's doc:
> When the heap size approaches max, V8 will perform series of garbage collections and invoke the [NearHeapLimitCallback](https://docs.rs/v8/latest/v8/struct.Isolate.html#method.add_near_heap_limit_callback). If the garbage collections do not help and the callback does not increase the limit, then V8 will crash with V8::FatalProcessOutOfMemory.

Reply:
The comment on the parameter is completely different from the one in the docs.

You actually missed the most important part of the docs: If the callback does not increase the limit, it will crash.

Reply:
The comment on the parameter is completely different from the one in the docs.

You actually missed the most important part of the docs: If the callback does not increase the limit, it will crash.